docs/user/gcp/install_upi.md

```diff
@@ -0,0 +1,507 @@
+# Install: User Provided Infrastructure (UPI)
+
+The steps for performing a UPI-based install are outlined here. Several
+[Deployment Manager][deploymentmanager] templates are provided to assist in
+completing these steps or to help model your own. You are also free to create
+the required resources through other methods; the templates are just an
+example.
+
+## Create Ignition Configs
+
+The machines will be started manually. Therefore, it is required to generate
+the bootstrap and machine Ignition configs and store them for later steps.
+Use [a staged install](../overview.md#multiple-invocations) to remove the
+control-plane Machines and compute MachineSets, because we'll be providing
+those ourselves and don't want to involve
+[the machine-API operator][machine-api-operator].
+
+```console
+$ openshift-install create install-config
+? SSH Public Key /home/user_id/.ssh/id_rsa.pub
+? Platform gcp
+? Project ID example-project
+? Region us-east1
+? Base Domain example.com
+? Cluster Name openshift
+? Pull Secret [? for help]
+```
+
+Edit the resulting `openshift-install.yaml` to set `replicas` to 0 for the `compute` pool:
+
+```console
+$ sed -i '1,/replicas: / s/replicas: .*/replicas: 0/' install-config.yaml
+```
+
+Create manifests to get access to the control-plane Machines and compute MachineSets:
+
+```console
+$ openshift-install create manifests
+INFO Consuming "Install Config" from target directory
+```
+
+From the manifest assets, remove the control-plane Machines and the compute MachineSets:
+
+```console
+$ rm -f openshift/99_openshift-cluster-api_master-machines-*.yaml openshift/99_openshift-cluster-api_worker-machinesets-*.yaml
+```
+
+You are free to leave the compute MachineSets in if you want to create compute
+machines via the machine API, but if you do you may need to update the various
+references (`subnet`, etc.) to match your environment.
+
+Now we can create the bootstrap Ignition configs.
+
+```console
+$ openshift-install create ignition-configs
+```
+
+After running the command, several files will be available in the directory.
+
+```console
+$ tree
+.
+├── auth
+│   └── kubeconfig
+├── bootstrap.ign
+├── master.ign
+├── metadata.json
+└── worker.ign
+```
+
+### Extract Infrastructure Name from Ignition Metadata
+
+By default, Ignition generates a unique cluster identifier comprised of the
+cluster name specified during the invocation of the installer and a short
+string known internally as the infrastructure name (infra_id). These values
```

Comment:
The parenthetical `(infra_id)` is not part of [the AWS docs][1].  I'd rather drop it, and just use "the infrastructure name" in text and `infraID` (and `INFRA_ID`, etc.) when we're actually interacting with it (e.g. extracting it from `metadata.json`).

[1]: https://github.com/openshift/installer/blame/40f7fa5ed322f06d37d7dfd1e8c222499b3be718/docs/user/aws/install_upi.md#L66

Reply:
Done.